File: skills/migrate-statsig/test-fixtures/phase2-examples/python-server/before/statsig_app.py
"""BEFORE - original Statsig usage (Statsig Python server SDK, `statsig`).

Migration INPUT. NOT part of the validated `src/` (Statsig deps not installed);
kept for the before/after diff and as a realistic transform input.

Resolve mode: Statsig server SDK = in-process eval. Target: Confidence Python
local-resolve provider (alpha) = in-process too -> NO resolve-mode change.
"""

import os
from dataclasses import dataclass

import statsig
from statsig import StatsigUser


@dataclass
class RequestUser:
    user_id: str
    email: str | None = None
    country: str | None = None
    plan: str | None = None


def handle_request(req: RequestUser) -> dict:
    statsig.initialize(os.environ["STATSIG_SERVER_SECRET"])

    user = StatsigUser(
        user_id=req.user_id,
        email=req.email,
        custom={"country": req.country, "plan": req.plan},
    )

    # Boolean feature gate.
    new_checkout = statsig.check_gate(user, "new_checkout")

    # Dynamic config - typed parameter reads with defaults.
    homepage = statsig.get_config(user, "homepage_config")
    title = homepage.get("title", "Welcome")
    max_items = homepage.get("maxItems", 10)

    # Experiment - typed parameter read with default.
    experiment = statsig.get_experiment(user, "checkout_button_experiment")
    button_color = experiment.get("buttonColor", "blue")

    return {
        "new_checkout": new_checkout,
        "title": title,
        "max_items": max_items,
        "button_color": button_color,
    }

File: skills/migrate-statsig/test-fixtures/phase2-examples/node-server/before/statsig-app.ts
// BEFORE — original Statsig usage (Statsig Server Core, @statsig/statsig-node-core).
//
// This file is the migration INPUT. It is intentionally NOT compiled (it's
// excluded from tsconfig and the Statsig deps aren't installed) — it exists so
// the `migrate-statsig plan code` flow has a realistic "before" to transform,
// and so the README before/after diff is grounded in real source.
//
// Resolve mode: Statsig server SDK = in-process eval (downloads project config,
// evaluates locally). Target: Confidence JS local-resolve = in-process too →
// NO resolve-mode change.

import { Statsig, StatsigUser } from '@statsig/statsig-node-core';

const statsig = new Statsig(process.env.STATSIG_SERVER_SECRET!);

let ready: Promise<void> | null = null;
function ensureReady(): Promise<void> {
  if (!ready) ready = statsig.initialize();
  return ready;
}

export type RequestUser = {
  userID: string;
  email?: string;
  country?: string;
  plan?: string;
};

export async function handleRequest(req: RequestUser) {
  await ensureReady();

  const user = new StatsigUser({
    userID: req.userID,
    email: req.email,
    // Statsig auto-derives country from IP server-side; here it's explicit.
    custom: { country: req.country, plan: req.plan },
  });

  // Boolean feature gate.
  const newCheckout = statsig.checkGate(user, 'new_checkout');

  // Dynamic config — typed parameter reads with defaults.
  const homepage = statsig.getDynamicConfig(user, 'homepage_config');
  const title = homepage.get('title', 'Welcome');
  const maxItems = homepage.get('maxItems', 10);

  // Experiment — typed parameter read with default.
  const experiment = statsig.getExperiment(user, 'checkout_button_experiment');
  const buttonColor = experiment.get('buttonColor', 'blue');

  return { newCheckout, title, maxItems, buttonColor };
}
